Source/Tools/ImageBaker.cpp

```diff
@@ -1050,7 +1052,7 @@ auto ImageBaker::LoadArt(string_view fname, string_view opt, File& file) -> Fram
             }
         }
         else {
-            dir_art = (dir + 1) % 8;
+            // dir_art = (dir + 1) % 8;
```

Comment:
It was crashing, because offset in SetCurPos has to start with dir_art = 0, and then go through all pictures in order 0 ... 7, and have correct FrameSize shifts.
The correct art orientation is then set here: (dir_art + 6) % 8

But sorry I just realized there is a mistake now for hexagonal case, I will update soon.

Reply:
Done

Reply:
In your case we iterate 8 times but fill only 6 dirs, so two times is unnecessary.
We must use
auto& sequence = dir == 0 ? collection.Main : collection.Dirs[dir - 1];
and fill dirs one by one.

Reply:
You have to iterate 8 times anyway to apply correct file position shifts, they have different FrameSize. Can only skip processing pixels

Reply:
This lines have to run for every direction to correctly calculate data offset:

file.SetCurPos(curpos);
curpos += frame_info.FrameSize;